clang/tools/clang-offload-wrapper/ClangOffloadWrapper.cpp

```diff
@@ -1083,10 +1108,55 @@ class BinaryWrapper {
           return FBinOrErr.takeError();
         Fbin = *FBinOrErr;
       } else {
-        Fbin = addDeviceImageToModule(
-            ArrayRef<char>(Bin->getBufferStart(), Bin->getBufferSize()),
-            Twine(OffloadKindTag) + Twine(ImgId) + Twine(".data"), Kind,
-            Img.Tgt);
+
+        // If '--offload-compress' option is specified and zstd is not available
+        // then warn the user that the image will not be compressed.
+        if (OffloadCompressDevImgs && !llvm::compression::zstd::isAvailable()) {
+          WithColor::warning(errs(), ToolName)
+              << "'--offload-compress' option is specified but zstd is not "
+                 "available. The device image will not be compressed.\n";
+        }
```

Comment:
Also, it would be great if we are consistent about handling error conditions.
Here we gracefully give out a warning message and use uncompressed image.
But, if compression fails below, we let SYCL crash.

Reply:
You're right. Looking at `clang-offload-bundler`, it throws an error when ZSTD and ZLIB are not available but the user still pass `--offload-compress` CLI option to clang (https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/OffloadBundler.cpp#L1519). We should do the same with `clang-offload-wrapper`.

Reply:
Fixed in https://github.com/intel/llvm/pull/15124/commits/1d8181335fb188aa4ae0ad39b3826a4162b200d2